cmd/codeqlExecuteScan.go

```diff
@@ -185,13 +187,46 @@ func uploadResults(config *codeqlExecuteScanOptions, repoInfo RepoInfo, token st
 
 	//if no git pramas are passed(commitId, reference, serverUrl, repository), then codeql tries to auto populate it based on git information of the checkout repository.
 	//It also depends on the orchestrator. Some orchestrator keep git information and some not.
+
+	var buffer bytes.Buffer
+	utils.Stdout(&buffer)
 	err := execute(utils, cmd, GeneralConfig.Verbose)
 	if err != nil {
 		log.Entry().Error("failed to upload sarif results")
-		return err
+		return "", err
 	}
+	utils.Stdout(log.Writer())
 
-	return nil
+	url := buffer.String()
+	return strings.TrimSpace(url), nil
+}
+
+func waitSarifUploaded(config *codeqlExecuteScanOptions, codeqlSarifUploader codeql.CodeqlSarifUploader) error {
+	var sarifUploadComplete = "complete"
+	var sarifUploadFailed = "failed"
+	maxRetries := config.SarifCheckMaxRetries
+	retryInterval := time.Duration(config.SarifCheckRetryInterval) * time.Second
+
+	log.Entry().Info("waiting for the SARIF to upload")
+	for i := 1; i <= maxRetries; i++ {
+		sarifStatus, err := codeqlSarifUploader.GetSarifStatus()
+		if err != nil {
+			log.Entry().Errorf("error while checking sarif status: %s, retrying in %d seconds... (retry %d/%d)", err, retryInterval, i, maxRetries)
+			time.Sleep(retryInterval)
+			continue
+		}
```

Comment:
502,503,504 should probably be retried. 500 is debatable - I'm inclined that not.

Reply:
well, the only error that could signal that it's worth retrying the request later, I think is a 503 code from the server. other errors will be repeated over and over again. I can suggest to process this code so that the request will be resubmitted only in 2 cases - if the file is still pending or the server is not available right now (this can be fixed during polling)